packages/eyes/src/generate-snapshots/generate-snapshots.ts

```diff
@@ -0,0 +1,107 @@
+const StylableWebpackPlugin = require('@stylable/webpack-plugin');
+import path from 'path';
+import webpack from 'webpack';
+import glob from 'glob';
+import {HTMLSnapshotPlugin} from '@stylable/webpack-extensions';
+import {createElement} from 'react';
+import {renderToStaticMarkup} from 'react-dom/server';
+import {generateFilteringLogic} from './filter-logic';
+import {mapSylesToComponents} from './map-styles-to-components';
+import {getCompName, IComponentMetadata, IRegistry} from '@ui-autotools/registry';
+import {consoleLog} from '@ui-autotools/utils';
+import {dedent} from './dedent';
+import {parseSnapshotFilename} from './filename-utils';
+
+function findComponentByName(name: string, Registry: IRegistry): IComponentMetadata<any> | void {
+    // We only have to do this because we currently map the component definitions to their metadata,
+    // not the names. So we have no way to get component metadata by name. And at this point in the build
+    // process, the component definition returned by webpack has been modified from the original, so we can't
+    // get the metadata with it
+  for (const component of Registry.metadata.components) {
+      const metadata = component[1];
+      if (getCompName(metadata.component) === name) {
+        return metadata;
+      }
+    }
+}
+
+async function buildSingleFile(fileName: string, filePath: string, directory: string, filteringLogic: (stylableModule: any) => any, config: any, Registry: IRegistry) {
+  function render(compiledFile: any, sourceFile: any) {
```

Comment:
Move `render` out of this function and provide `fileName` either by binding it:
```
new HTMLSnapshotPlugin({
   render: render.bind(null, fileName)
})
```
or by defining an anonymous function:
```
new HTMLSnapshotPlugin({
  render: (...args) => render(fileName, ..args)
})
```
Do the same for `filteringLogic`